yt/frontends/ramses/fields.py

```diff
@@ -192,6 +198,35 @@ def _temperature(field, data):
         if rt_flag: # rt run
             self.create_rt_fields()
 
+        #Load magnetic fields
+        if ('gas','magnetic_field_x_left') in self:
+            self.create_magnetic_fields()
+
+    def create_magnetic_fields(self):
+        #Calculate cell-centred magnetic fields from face-centred
+        def mag_field(ax):
+            def _mag_field(field, data):
+                return (data['magnetic_field_%s_left' % ax]+data['magnetic_field_%s_right' % ax])/2
+            return _mag_field
+
+        for ax in self.ds.coordinates.axis_order:
+            self.add_field(('gas',"magnetic_field_%s" % ax),
+                           sampling_type='cell',
+                           function=mag_field(ax),
+                           units=self.ds.unit_system['magnetic_field_cgs'])
+
+        def _divB(field,data):
+            '''Calculate magnetic field divergence'''
+            out=data['zeros']
```

Comment:
With my comments below, I am afraid this may result in `data['zeros']` being updated in place. Maybe 
```suggestion
            out = np.zeros_like(data['magnetic_field_x_right'])
``` 
would be better?